custom/enikshay/views.py

```diff
@@ -108,3 +108,34 @@ def run_task(task_name):
         elif task_requested in ReconciliationTaskForm.permitted_tasks:
             run_task(task_requested)
         return JsonResponse({'message': 'Task queued. You would get an email shortly.'})
+
+
+class DataDumpTaskView(TemplateView):
+    template_name = "enikshay/data_dumps_task.html"
+
+    @method_decorator(require_superuser)
```

Comment:
If you decorate the `dispatch` method, it'll take effect for all request methods (ie `get` and `post`), since `dispatch` is the primary entry point, which then directs to the appropriate method.  (Quick plug for [ccbv.co.uk](http://ccbv.co.uk/projects/Django/1.11/django.views.generic.base/TemplateView/), which is the best way I've seen to browse the source of Django class-based views).

Also, you can use `@method_decorator` on the class itself by specifying the method you which to decorate:
```python
@method_decorator(require_superuser, name='dispatch')
class DataDumpTaskView(TemplateView):
    ...
```
This is nice so you don't have to override `dispatch` and call `super` just to decorate it.